unittests/test_iem.py

```diff
@@ -0,0 +1,111 @@
+        }
+        EventMessage.init(component='sspl', source='S')
+
+    def mock_ipmitool_fault(self):
+        cmd = "yum remove ipmitool"
+        run_command(cmd)
+        event = self.EVENT_CODE['IPMITOOL_ERROR']
+        self.module = event[1]
+        self.event_code = event[0]
+
+    def mock_ipmitool_fault_resolved(self):
+        cmd = "yum install ipmitool"
+        run_command(cmd)
+        event = self.EVENT_CODE['IPMITOOL_AVAILABLE']
+        self.module = event[1]
+        self.event_code = event[0]
+
+    def test_01_ipmitool_fault_alert_send(self):
+        """ Test iem 'ipmitool' fault alert & send it to MessageBus """
+
+        print("### TestCase: test_01_ipmitool_fault_alert_send\n")
+        self.mock_ipmitool_fault()
+        severity = 'E'
+        description = "ipmitool command execution error"
+        EventMessage.send(module=self.module, event_id=self.event_code,
+                              severity=severity, message_blob=description)
+
+    def test_02_ipmitool_fault_alert_receive(self):
+        """ Test iem 'ipmitool' fault alert receive """
+
+        print("### TestCase: test_02_ipmitool_fault_alert_receive\n")
+        EventMessage.subscribe(component='sspl')
+        fault_alert = EventMessage.receive()
+        print(f"IEM Received:{fault_alert}")
+        self.assertIs(type(fault_alert), dict)
+        self.assertEqual(fault_alert["iem"]["info"]["severity"], "Error")
+        self.assertEqual(fault_alert["iem"]["source"]["module"], "ipmitool")
+        self.assertEqual(fault_alert["iem"]["contents"]["event"], "0050010001")
+
+    def test_03_ipmitool_fault_resolved_alert_send(self):
+        """ Test iem 'ipmitool' fault_resolved alert and
+        send it to MessageBus """
+
+        print("### TestCase: test_03_ipmitool_fault_resolved_alert_send\n")
+        self.mock_ipmitool_fault_resolved()
+        severity = 'I'
+        description = "ipmitool command execution success again."
+        EventMessage.send(module=self.module, event_id=self.event_code,
+                              severity=severity, message_blob=description)
+
+    def test_04_ipmitool_fault_resolved_receive(self):
+        """ Test iem 'ipmitool' fault_resolved alert receive """
+
+        print("### TestCase: test_04_ipmitool_fault_resolved_receive\n")
+        EventMessage.subscribe(component='sspl')
+        fault_alert = EventMessage.receive()
+        print(f"IEM Received:{fault_alert}")
+        self.assertIs(type(fault_alert), dict)
+        self.assertEqual(fault_alert["iem"]["info"]["severity"], "Informational")
+        self.assertEqual(fault_alert["iem"]["source"]["module"], "ipmitool")
+        self.assertEqual(fault_alert["iem"]["contents"]["event"], "0050010002")
+
+    def tearDown(self):
+        pass
+
+def run_command(command):
+    """Run the command and get the response and error returned"""
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
`shell=True` is platform dependent and prone to shell injection, please avoid using it.

Reply:
@jjxsg run_command func is getting used in other files like( disk_monitor.py, raid.py, disk_msg_handler.py etc) 
in the same manner.
Also I tried executing cmd without shell=True param, but it throws error.
Do you have any workaround?

Reply:
@gargpalak048 please refer to first comment in this PR https://github.com/Seagate/cortx-monitor/pull/404
You need to split the command properly using `shlex` to be able to execute command without starting a shell.

Reply:
Thanks for the ref @jjxsg.
So in cortx-utils SimpleProcess class, this func already implemented.
Just parse the cmd and it will return the response.
No need to shlex here.

Reply:
@gargpalak048 Sorry, I didn't get you. Which function you are talking about? And I can see `subprocess` with `shell=True`

Reply:
Using SimpleProcess.run() instead of run_command() func.
Will remove this func from test script